cmd/fleet/cron.go

```diff
@@ -745,11 +751,68 @@ func newCleanupsAndAggregationSchedule(
 				return ds.UpdateOSVersions(ctx)
 			},
 		),
+		schedule.WithJob(
+			"verify_disk_encryption_keys",
+			func(ctx context.Context) error {
+				return verifyDiskEncryptionKeys(ctx, logger, ds, config)
+			},
+		),
```

Comment:
@mna @gillespi314 could you help me out with this? 🙏 I'd like to hear your opinions on this approach.

We need to pre-compute if we're able to decrypt the key, as this boolean will be used in the host detail response and as a filter in the host lists endpoint.

I have considered spawning a [worker job](https://github.com/fleetdm/fleet/tree/main/server/worker) in `directIngestDiskEncryptionKeyDarwin` but I wasn't convinced.

Encryption keys don't change often, and they take time to be generated (requires the user logging out), but there's at least one scenario when we will get a lot of new keys to verify: during MDM migration. The locking shouldn't be too bad for the numbers we're considering, but I wonder if we should consider another approach or batch the `UPDATE`s.

Reply:
If we can live with the slight delay that is implied with the cron job, I prefer this approach. Host ingestion is super performance-sensitive, and spawning "launch-and-forget" goroutines is best to be avoided, to be honest - it's a bit of a debugging/maintenance nightmare, is not taken into account by e.g. graceful server shutdowns, can block/leak/panic/etc. EDIT: sorry, just saw you meant the worker as we use for jira/zendesk tickets. Yeah that could work too but again, if the cron approach is acceptable it might still be best as we can do more batchy work (instead of creating a worker entry for each single host key and process them one at a time).

Reply:
I'm very interested to hear Martin's thoughts. It seems like something we should try in load testing assuming that our target service level of 10k MDM hosts are all migrating with encryption keys to be verified during the same cron interval.

Reply:
I think a delay is totally fine considering that the user action required to get an encrypted key (logging out) can take days.

Load testing makes sense too, but if we're fine with this I would rather spend those cycles (as it will require a bit of juggling with the SCEP keys and the whole MDM setup) on more pressing stuff. Note that setting up an MDM load test is something that we will have to do eventually! I'm just advocating for pushing it down the road.

Reply:
> I think a delay is totally fine considering that the user action required to get an encrypted key (logging out) can take days.

:+1: 

> Note that setting up an MDM load test is something that we will have to do eventually! I'm just advocating for pushing it down the road.

Totally. It can be a big time sink and will be even more so to get the MDM stuff properly setup. Some napkin nap on my end, assuming 300ms by decryption (not sure if this is similarly slow to e.g. modern password encryption), it'd be tight but should fit about 10K in an hour. But I don't think it's an issue if it doesn't? Or is it because the job's context will be cancelled and by the time we try to save the results, the updates would fail?

Reply:
lol ~nap~ math :D

Reply:
From what I can see, the `schedule` will actually extend the lock if the jobs run for longer? It doesn't look like it would fail if for some reason the job ran for more than the 1 hour allocated, AFAICS (feel free to double-check me and correct me if I'm wrong, took a ~quick look and it's not code I'm familiar with).

Reply:
That's right. The `schedule` allows for runs that take longer than the scheduled interval.